src/main/java/de/rototor/pdfbox/graphics2d/RGBtoCMYKColorMapper.java

```diff
@@ -0,0 +1,58 @@
+package de.rototor.pdfbox.graphics2d;
+
+import org.apache.pdfbox.pdmodel.PDPageContentStream;
+import org.apache.pdfbox.pdmodel.graphics.color.PDColor;
+import org.apache.pdfbox.pdmodel.graphics.color.PDDeviceCMYK;
+
+import java.awt.*;
+import java.awt.color.ICC_ColorSpace;
+import java.awt.color.ICC_Profile;
+
+/*
+  Usage:
+
+  PdfBoxGraphics2D pdfBoxGraphics2D = new PdfBoxGraphics2D(this.doc, (int)(width), (int)(height));
+  PdfBoxGraphics2DColorMapper colorMapper = new RGBtoCMYKColorMapper(icc_profile);
+  pdfBoxGraphics2D.setColorMapper(colorMapper);
+
+  Where icc_profile is an instance of java.awt.color.ICC_Profile that supports a CMYK
+  colorspace.  For testing purposes, we're using ISOcoated_v2_300_bas.icc which ships
+  with PDFBox.
+ */
+
+
+public class RGBtoCMYKColorMapper extends PdfBoxGraphics2DColorMapper {
+    ICC_ColorSpace icc_colorspace;
+
+    public RGBtoCMYKColorMapper(ICC_Profile icc_profile){
+        resetColorSpace(icc_profile);
+    }
+
+    public void resetColorSpace(ICC_Profile icc_profile){
+        icc_colorspace = new ICC_ColorSpace(icc_profile);
```

Comment:
You should construct a PDICCBased() based on this profile. i.e.
```
					this.pdProfile = new PDICCBased(document);
					OutputStream outputStream = pdProfile.getPDStream().createOutputStream(COSName.FLATE_DECODE);
					outputStream.write(profile.getData());
					outputStream.close();
					pdProfile.getPDStream().getCOSObject().setInt(COSName.N, 4);
					pdProfile.getPDStream().getCOSObject().setItem(COSName.ALTERNATE, COSName.DEVICECMYK);
```

To do so you of course need the PDDocument.

Reply:
Updated.